### PR TITLE
Handle pathlib.Path objects

### LINKED ISSUE #6652: Handle pathlib paths with `get_file_obj`
#### Is your feature request related to a problem? Please describe
I'm trying to read a statsmodels.tsa.statespace.sarimax.SARIMAXResults pickle file I've saved. I'd like to pass a pathlib.Path object as I find it easier to handle (I work with different machines, Mac and Windows).

#### Describe the solution you'd like
The function `open` already supports pathlib objects. Why not using it?

You could replace `statsmodels.iolib.openfile.get_file_obj` with:

```python
if _is_string_like(fname):
    return _open(fname, mode, encoding)
try:  # try opening it with 'open'
    return open(fname, mode, encoding)
except TypeError:  # else, maybe return context manager (please check if it's always TypeError)
    try:
        # Make sure the object has the write methods
        if 'r' in mode:
            fname.read
        if 'w' in mode or 'a' in mode:
            fname.write
    except AttributeError:
        raise ValueError('fname must be a string or a file-like object')
    return EmptyContextManager(fname)
```

This is just a suggestion though :)

### PR DESCRIPTION
**Pull request by giuliobeseghi**
- [ ] closes #6652 
- [ ] tests added / passed.

### CODE REVIEW

**Comment by bashtage** [comment]
The failures are not yours.  This is working.

File: statsmodels/iolib/tests/test_pickle.py [hunk 1]
```diff
@@ -8,16 +9,26 @@
...
-    b = load_pickle(tmpdir+'/res.pkl')
+
+    # test with str
+    path_str = tmpdir + "/res.pkl"
+    save_pickle(a, path_str)
+    b = load_pickle(path_str)
+    assert_equal(a, b)
+
+    # test with pathlib
+    path_pathlib = pathlib.Path(tmpdir) / "res.pkl"
```
Comment (bashtage) [inline statsmodels/iolib/tests/test_pickle.py:22]: Maybe use a different file name just to be sure it is writing the file. res2.pkl.

File: statsmodels/iolib/tests/test_pickle.py [hunk 2]
```diff
@@ -8,16 +9,26 @@
...
+    save_pickle(a, path_pathlib)
+    b = load_pickle(path_pathlib)
     assert_equal(a, b)
 
     # cleanup, tested on Windows
     try:
         import os
-        os.remove(tmpdir+'/res.pkl')
+
+        os.remove(tmpdir + "/res.pkl")
```
Comment (bashtage) [inline statsmodels/iolib/tests/test_pickle.py:31]: Need to remove both here too then.

**Comment by giuliobeseghi** [comment]
Also, we could split `test_pickle` to three different tests:

- test_pickle_from_string
- test_pickle_from_path_obj
- test_pickle_from_file_handler

**Comment by bashtage** [comment]
I'm happy with the way it is; best would be to use `@pytest.mark.parameterize("path",[<a string>, <a Pathlib>, <a fh>])`.  Happy if you want to do this or not.

**Comment by bashtage** [comment]
I missed one. Could you update the docstring here:

https://github.com/statsmodels/statsmodels/pull/6654/files#diff-bdb9c768dd6d4c03d15a5015c523a710R43

**Comment by bashtage** [comment]
This is line 43, which is above your changes.

**Comment by giuliobeseghi** [comment]
> I'm happy with the way it is; best would be to use `@pytest.mark.parameterize("path",[<a string>, <a Pathlib>, <a fh>])`. Happy if you want to do this or not.

We'd have to define the pathlib obj and the fh outside (maybe at the top of the module). Then we'd have to add an if/else block to handle either str and pathlib objs or fhs in two different ways. I think it's fine as it is now (although if the test fails, it fails for the three of them together).

**Comment by bashtage** [comment]
Thanks for this.

**Pull request merged**